js/src/ui/TxList/store.js

```diff
@@ -0,0 +1,99 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { action, observable, transaction } from 'mobx';
+
+export default class Store {
+  @observable blocks = {};
+  @observable sortedHashes = [];
+  @observable transactions = {};
+
+  constructor (api) {
+    this._api = api;
+  }
+
+  @action addBlocks = (blocks) => {
+    this.blocks = Object.assign({}, this.blocks, blocks);
+  }
+
+  @action addTransactions = (transactions) => {
+    transaction(() => {
+      this.transactions = Object.assign({}, this.transactions, transactions);
+      this.sortedHashes = Object
+        .keys(this.transactions)
+        .sort((ahash, bhash) => {
+          const bnA = this.transactions[ahash].blockNumber;
+          const bnB = this.transactions[bhash].blockNumber;
+
+          return bnB.comparedTo(bnA);
+        });
+    });
+  }
+
+  loadTransactions (_txhashes) {
```

Comment:
Could be. Technically the only updates happen in the `addTransactions` function. So `@action` won't hurt, however the function itself doesn't update the object state. (Needed to split the change out into a different `@action` - async, so doesn not play well with `setStrict`)